Executables/Game/Extensions/ItemExtensions.cs

```diff
@@ -105,7 +100,7 @@ public static uint GetMaxMagicWeaponDamage(this ItemData item)
         return null;
     }
 
-    public static async Task<ItemInstance> GetItem(this IDbConnection db, ICacheManager cacheManager, Guid id)
+    public static async Task<ItemInstance?> GetItem(this IItemRepository repository, ICacheManager cacheManager, Guid id)
```

Comment:
As an overarching architectural idea: Maybe this method could move to `ICacheManager` which in turn knows `IItemRepository`. The happy pass of this is that the cache already contains the requested item with DB lookup being the fallback. Would eliminate the need for this extension method (and the one below if treated the same).

This would make the CacheManager a little more self-contained imo

Reply:
Yes your are right with this idea. However this is out of scope of this PR. There is another PR for this however: #13